config/config.go

```diff
@@ -59,10 +72,58 @@ func Init(flags *pflag.FlagSet) {
 		if flag.Name == "help" {
 			return
 		}
-		viper.Set(flag.Name, flag.Value.String())
+
+		// Only updated parsed configs if the user sets value
+		if flag.Changed {
+			viper.Set(flag.Name, flag.Value.String())
+		}
 	})
 
 	if err := viper.Unmarshal(&Config); err != nil {
 		panic(err)
 	}
 }
+
+// ConfigureParentDirPaths Creates the default parent directory which can be used for logs and persistent data
+func configureParentDirPaths() {
+	var err error
+	DefaultParentDir, err = createDefaultParentDir()
+	if err != nil {
+		slog.Warn("Failed to create default preferences directory", slog.String("error", err.Error()))
+	}
+}
+
+func createDefaultParentDir() (string, error) {
+	// Get the default directory path for the OS
+	prefDir := GetDefaultParentDirPath()
+	if err := os.MkdirAll(prefDir, os.ModePerm); err != nil {
+		return "", fmt.Errorf("failed to create preferences directory: %w", err)
+	}
+
+	return prefDir, nil
+}
```

Comment:
this logic will become much simpler once we have OS specific files with the same variable name. Ref comment - https://github.com/golang/go/issues/18364#issue-196240462.